qiita_db/support_files/patches/33.sql

```diff
@@ -0,0 +1,435 @@
+ ) ;
+CREATE INDEX idx_study_artifact_study ON qiita.study_artifact ( study_id ) ;
+CREATE INDEX idx_study_artifact_artifact ON qiita.study_artifact ( artifact_id ) ;
+ALTER TABLE qiita.study_artifact ADD CONSTRAINT fk_study_artifact_study FOREIGN KEY ( study_id ) REFERENCES qiita.study( study_id )    ;
+ALTER TABLE qiita.study_artifact ADD CONSTRAINT fk_study_artifact_artifact FOREIGN KEY ( artifact_id ) REFERENCES qiita.artifact( artifact_id )    ;
+
+-- Software table - holds the information of a given software package present
+-- in the system and can be used to process an artifact
+CREATE TABLE qiita.software (
+    software_id          bigserial  NOT NULL,
+    name                 varchar  NOT NULL,
+    version              varchar  NOT NULL,
+    description          varchar  NOT NULL,
+    CONSTRAINT pk_software PRIMARY KEY ( software_id )
+ ) ;
+
+-- soft_command table - holds the information of a command in a given software
+-- this table should be renamed to command once the command table in the
+-- analysis table is merged with this one
+CREATE TABLE qiita.soft_command (
+    command_id           bigserial  NOT NULL,
+    name                 varchar  NOT NULL,
+    software_id          bigint  NOT NULL,
+    description          varchar  NOT NULL,
+    cli_cmd              varchar  ,
+    parameters_table     varchar  NOT NULL,
+    CONSTRAINT pk_soft_command PRIMARY KEY ( command_id )
+ ) ;
+CREATE INDEX idx_soft_command ON qiita.soft_command ( software_id ) ;
+ALTER TABLE qiita.soft_command ADD CONSTRAINT fk_soft_command_software FOREIGN KEY ( software_id ) REFERENCES qiita.software( software_id );
+
+-- Publication table - holds the minimum information for a given publication
+-- It is useful to keep track of the publication of the studies and the software
+-- used for processing artifacts
+CREATE TABLE qiita.publication (
+    doi                  varchar  NOT NULL,
+    pubmed_id            integer  ,
+    CONSTRAINT pk_publication PRIMARY KEY ( doi )
+ ) ;
+
+-- Software publictation table - relates each software package with the lists of
+-- its related publciations
+CREATE TABLE qiita.software_publication (
+    software_id          bigint  NOT NULL,
+    publication_doi      varchar  NOT NULL,
+    CONSTRAINT idx_software_publication_0 PRIMARY KEY ( software_id, publication_doi )
+ ) ;
+CREATE INDEX idx_software_publication_software ON qiita.software_publication ( software_id ) ;
+CREATE INDEX idx_software_publication_publication ON qiita.software_publication ( publication_doi ) ;
+ALTER TABLE qiita.software_publication ADD CONSTRAINT fk_software_publication FOREIGN KEY ( software_id ) REFERENCES qiita.software( software_id )    ;
+ALTER TABLE qiita.software_publication ADD CONSTRAINT fk_software_publication_0 FOREIGN KEY ( publication_doi ) REFERENCES qiita.publication( doi )    ;
+
+-- Add remaining FK to the artifact table. Creating here since the target tables
```

Comment:
Good point - I was biased by the order in what dbschema generated the stuff. I re-ordered the table creation.